File: src/api/dto.ts
export interface PostListDTO {
  id: number;
  title: string;
  content: string;
  createdAt: string;
  views: number;
  tags: string[];
  commentCount: number;
  url?: string;
  thumbnailUrl?: string;
}

export interface PostDetailDTO extends PostListDTO {
  comments: CommentDTO[];
}

export interface CommentDTO {
  id: number;
  postId: number;
  content: string;
  createdAt: string;
}

export interface PostCreateDTO {
  title: string;
  content: string;
  tags?: string[];
  url?: string;
  thumbnailUrl?: string;
  password: string;
}

export interface PostUpdateDTO {
  title: string;
  content: string;
  tags?: string[];
  url?: string;
  thumbnailUrl?: string;
}

export interface UrlAnalysisData {
  url: string;
  title: string;
  summary: string[];
  tags: string[];
}

File: backend/dto.py
from pydantic import BaseModel
from typing import List, Optional

class PostCreate(BaseModel):
    title: str
    content: str
    tags: List[str] = []
    url: Optional[str] = None
    thumbnailUrl: Optional[str] = None
    password: str

class PostUpdate(BaseModel):
    title: str
    content: str
    tags: List[str]
    url: Optional[str] = None
    thumbnailUrl: Optional[str] = None

class CommentCreate(BaseModel):
    postId: int
    content: str
    password: str 